lib/MiniDroneBtAdapter.js

```diff
@@ -305,16 +400,26 @@ class MiniDroneBtAdapter extends EventEmitter {
         if (!peripheral) {
             return false;
         }
-
         const localName = peripheral.advertisement.localName;
         const manufacturer = peripheral.advertisement.manufacturerData;
-        const localNameMatch = localName && DRONE_PREFIXES.some((prefix) =>
-            localName.indexOf(prefix) >= 0
-        );
-        const manufacturerMatch = manufacturer && (MANUFACTURER_SERIALS.indexOf(manufacturer) >= 0);
 
-        // Is true for EITHER an "RS_" name OR manufacturer code.
-        return localNameMatch || manufacturerMatch;
+        if (this.options.droneFilter == '') {
+          const localNameMatch = localName && DRONE_PREFIXES.some((prefix) =>
+              localName.indexOf(prefix) >= 0
+          );
+          const manufacturerMatch = manufacturer && (MANUFACTURER_SERIALS.indexOf(manufacturer) >= 0);
+
+          // Is true for EITHER an "RS_" name OR manufacturer code.
+          return localNameMatch || manufacturerMatch;
+        }
+        else {
+          if (localName == this.options.droneFilter) {
+            return true;
+          }
+          else {
+            return false;
+          }
+        }
```

Comment:
validatePeripheral(peripheral) {
        if (!peripheral) {
            return false;
        }
        const localName = peripheral.advertisement.localName;
        const manufacturer = peripheral.advertisement.manufacturerData;
        const localNameMatch = localName && DRONE_PREFIXES.some((prefix) =>
            localName.indexOf(prefix) >= 0
        );
        const manufacturerMatch = manufacturer && (MANUFACTURER_SERIALS.indexOf(manufacturer) >= 0);

        if (this.options.droneFilter && localName === this.options.droneFilter) {
            return true;
        }

        // Is true for EITHER an "RS_" name OR manufacturer code.
        return localNameMatch || manufacturerMatch;
    }